elasticapm/transport/base.py

```diff
@@ -97,14 +98,16 @@ def __init__(
         self._flushed = threading.Event()
         self._closed = False
         self._processors = processors if processors is not None else []
-        # only start the event processing thread if we are not in a uwsgi master process
-        if not is_master_process():
-            self._start_event_processor()
-        else:
-            # if we _are_ in a uwsgi master process, use the postfork mixup to start the thread after the fork
-            compat.postfork(lambda: self._start_event_processor())
+        self._pid = None
+        self._thread_starter_lock = threading.Lock()
 
     def queue(self, event_type, data, flush=False):
+        with self._thread_starter_lock:
+            # check if self._pid is the same as os.getpid(). If they are not the same, it means that our
+            # process was forked at some point, so we need to start another processor thread
+            if self._pid != os.getpid():
+                self._start_event_processor()
```

Comment:
Since the `_start_event_processor()` already has the check

`if (not self._event_process_thread or not self._event_process_thread.is_alive()) and not self._closed:`

would it not be sufficient just to have it, instead of the pid checking.

Or if pid check is more reliable, in case a dispatcher can fork an already running process, then should this check from the `_start_event_processor` be removed?